src/renderer/services/wallets.js

```diff
@@ -120,50 +263,291 @@ export default {
   },
 
   openEncryptedKey(encryptedKey, passphrase) {
-    return new Promise((resolve, reject) => {
+    console.log('enc called');
+    return new Promise(async (resolve, reject) => {
       try {
         const wif = wallet.decrypt(encryptedKey, passphrase);
         const account = new wallet.Account(wif);
+        const net = network.getSelectedNetwork().net;
+
+        let mnemonic;
+        let btcTestnetAddress;
+        let btcMainnetAddress;
+        let btcTestnetWalletClient;
+        let btcMainnetWalletClient;
+        let btcKey;
+
+        if (storageNew.walletExists(wif)) {
+          const walletToOpen = storageNew.getOne(wif);
+          const mnemonicBytes = CryptoJS.AES.decrypt(walletToOpen.encryptedMnemonicString, wif);
+          mnemonic = mnemonicBytes.toString(CryptoJS.enc.Utf8);
+          const dataBTCgen = await this.getBTCWalletObjects(mnemonic);
+          if (dataBTCgen.err) {
+            return reject('An error occured while trying to login to Bitcoin wallet.');
+          }
+          btcKey = dataBTCgen.key;
+          btcTestnetWalletClient = dataBTCgen.btcTestnetWalletClient;
+          btcMainnetWalletClient = dataBTCgen.btcMainnetWalletClient;
+          btcTestnetAddress = walletToOpen.btcTestnetAddress;
+          btcMainnetAddress = walletToOpen.btcMainnetAddress;
+        } else {
+          const dataBTCgen = await this.generateBTCETHWallet('wall');
```

Comment:
This should be in a method for generating a new wallet, this is duplicated code from elsewhere.

Reply:
it is already a method